agent/azure-monitor-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/MetricDataMapper.java

```diff
@@ -71,62 +64,53 @@ public AzureMonitorMetricExporter(TelemetryClient telemetryClient) {
   }
 
-  @Override
-  public AggregationTemporality getAggregationTemporality(InstrumentType instrumentType) {
-    return AggregationTemporalitySelector.deltaPreferred()
-        .getAggregationTemporality(instrumentType);
+  public MetricDataMapper(
+      String instrumentationKey,
+      @Nullable Consumer<AbstractTelemetryBuilder> telemetryInitializer) {
+    this.instrumentationKey = instrumentationKey;
+    this.telemetryInitializer = telemetryInitializer;
   }
 
-  @Override
-  public CompletableResultCode export(Collection<MetricData> metrics) {
-    if (stopped.get()) {
-      return CompletableResultCode.ofFailure();
+  public void map(MetricData metricData, Consumer<TelemetryItem> consumer) {
+    if (EXCLUDED_METRIC_NAMES.contains(metricData.getName())) {
+      return;
     }
 
-    for (MetricData metricData : metrics) {
-      if (EXCLUDED_METRIC_NAMES.contains(metricData.getName())) {
-        continue;
-      }
-
-      MetricDataType type = metricData.getType();
-      if (type == DOUBLE_SUM
-          || type == DOUBLE_GAUGE
-          || type == LONG_SUM
-          || type == LONG_GAUGE
-          || type == HISTOGRAM) {
-        convertOtelMetricToAzureMonitorMetric(metricData).forEach(telemetryClient::trackAsync);
-      } else {
-        logger.warn("metric data type {} is not supported yet.", type);
+    MetricDataType type = metricData.getType();
+    if (type == DOUBLE_SUM
+        || type == DOUBLE_GAUGE
+        || type == LONG_SUM
+        || type == LONG_GAUGE
+        || type == HISTOGRAM) {
+      List<TelemetryItem> telemetryItemList = convertOtelMetricToAzureMonitorMetric(metricData);
+      for (TelemetryItem telemetryItem : telemetryItemList) {
+        consumer.accept(telemetryItem);
       }
+    } else {
+      logger.warn("metric data type {} is not supported yet.", metricData.getType());
     }
-    return CompletableResultCode.ofSuccess();
-  }
-
-  @Override
-  public CompletableResultCode flush() {
-    return CompletableResultCode.ofSuccess();
-  }
-
-  @Override
-  public CompletableResultCode shutdown() {
-    stopped.set(true);
-    return CompletableResultCode.ofSuccess();
   }
 
   private List<TelemetryItem> convertOtelMetricToAzureMonitorMetric(MetricData metricData) {
     List<TelemetryItem> telemetryItems = new ArrayList<>();
-
     for (PointData pointData : metricData.getData().getPoints()) {
-      MetricTelemetryBuilder builder = telemetryClient.newMetricTelemetryBuilder();
+      MetricTelemetryBuilder builder = MetricTelemetryBuilder.create();
+      if (telemetryInitializer != null) {
+        telemetryInitializer.accept(builder);
+      }
+      builder.setInstrumentationKey(instrumentationKey);
+      builder.addTag(
+          ContextTagKeys.AI_INTERNAL_SDK_VERSION.toString(), VersionGenerator.getSdkVersion());
       builder.setTime(FormattedTime.offSetDateTimeFromEpochNanos(pointData.getEpochNanos()));
+      builder.setSampleRate(100); // TODO make this configurable?
```

Comment:
I think maybe we should not set this, can you check what we were doing previously?

```suggestion
```